site/content/extend/plugins/developer-setup.md

```diff
@@ -0,0 +1,54 @@
+---
+title: "Developer Setup"
+date: 2020-07-11T23:00:00-04:00
+weight: 2
+subsection: Plugins
+---
+
+Once you have your [server](/contribute/server/developer-setup) and [webapp](/contribute/webapp/developer-setup) set up, you can start developing on plugins.
+
+For developing on Mattermost-managed plugins, each plugin's setup instructions can be found in the plugin's repository's README. Some plugins do not have external dependencies and require little to no setup, like the [Todo Plugin](https://github.com/mattermost/mattermost-plugin-todo) while others require an external service to be set up, like the [Jira Plugin](https://github.com/mattermost/mattermost-plugin-jira) and [GitHub Plugin](https://github.com/mattermost/mattermost-plugin-github).
+
+## Set up your environment to deploy plugins
+
+### Deploying with Local Mode
+
+If your Mattermost server is running locally, you can enable [local mode](https://docs.mattermost.com/administration/mmctl-cli-tool.html#local-mode) to streamline deploying your plugin. Edit your server configuration as follows:
```

Comment:
That will only work for plugin freshly synced with the starter-template. Should this caveat be noted?

Reply:
Did this comment got addressed?

Reply:
Addressed now :+1: